chain/client/src/client_actor.rs

```diff
@@ -1381,8 +1381,13 @@ impl ClientActorInner {
         };
 
         // If we produced the optimistic block, send it out before we save it.
+        let tip = self.client.chain.head()?;
+        let targets = self.client.get_optimistic_block_targets(&tip)?;
         self.network_adapter.send(PeerManagerMessageRequest::NetworkRequests(
-            NetworkRequests::OptimisticBlock { optimistic_block: optimistic_block.clone() },
+            NetworkRequests::OptimisticBlock {
+                chunk_producers: (*targets).clone(),
+                optimistic_block: optimistic_block.clone(),
```

Comment:
What do you think about wrapping those in `Arc`s to avoid the cloning?

Reply:
I made the change for `chunk_producers`. For `optimistic_block` further changes are needed on the network side to avoid the clone so I'm leaving it for now.